File: apps/fago_app/lib/screens/touro_screen.dart
import 'package:flutter/material.dart';
import '../services/whatsapp_service.dart';
import '../services/location_service.dart';
import '../features/profile/services/profile_service.dart';
import 'web_module_screen.dart';

class TourOScreen extends StatefulWidget {
  const TourOScreen({super.key});

  @override
  State<TourOScreen> createState() => _TourOScreenState();
}

class _TourOScreenState extends State<TourOScreen> {
  String _selectedPackage = 'arupadaiveedu';
  int _passengersCount = 4;

  final Map<String, Map<String, dynamic>> _tourPackages = {
    'arupadaiveedu': {
      'title': 'அறுபடைவீடு ஆன்மீக பயணம் (Arupadaiveedu Murugan Circuit)',
      'subtitle': 'Palani, Tiruchendur, Swamimalai, Thiruthani, Madurai',
      'icon': '🕉️',
      'baseRate': 12500.0,
      'duration': '3 Days / 2 Nights',
      'vehicle': 'Innova / Ertiga / Tempo Traveller',
    },
    'rameswaram': {
      'title': 'இராமேஸ்வரம் & கன்னியாகுமரி தரிசனம் (Rameswaram & Kanyakumari)',
      'subtitle': 'Ramanathaswamy Temple, Dhanushkodi, Vivekananda Rock',
      'icon': '🌊',
      'baseRate': 14000.0,
      'duration': '3 Days / 2 Nights',
      'vehicle': 'AC SUV / Force Traveller',
    },
    'ooty_kodai': {
      'title': 'ஊட்டி & கொடைக்கானல் மலை சுற்றுலா (Ooty & Kodaikanal Hills)',
      'subtitle': 'Botanical Garden, Pykara, Pillar Rocks, Coaker Walk',
      'icon': '⛰️',
      'baseRate': 16500.0,
      'duration': '4 Days / 3 Nights',
      'vehicle': 'Ghat Road Expert Driver + Force Traveller',
    },
    'tanjore_chola': {
      'title': 'தஞ்சை பெரிய கோவில் & சோழ மண்டலம் (Tanjore Big Temple Circuit)',
      'subtitle': 'Brihadeeswarar Temple, Gangaikonda Cholapuram, Darasuram',
      'icon': '🛕',
      'baseRate': 8500.0,
      'duration': '2 Days / 1 Night',
      'vehicle': 'Sedan / SUV / Tempo',
    },
  };

  Future<void> _bookTourPackage() async {
    final loc = await LocationService().getCurrentLocation();
    final pinData = await LocationService().getPincodeAndAddressFromCoordinates(loc.latitude, loc.longitude);
    final userDetails = await ProfileService.getCurrentUserProfileDetails();
    final userName = userDetails['name'] ?? '';

    final pkg = _tourPackages[_selectedPackage]!;
    final StringBuffer sb = StringBuffer();
    sb.writeln('🕉️ *TOURO TAMIL NADU TEMPLE & HILL TOUR BOOKING* 🕉️\n');
    if (userName.isNotEmpty) {
      sb.writeln('👤 *Customer Name*: $userName');
    }
    sb.writeln('🚩 *Selected Package*: ${pkg['title']}');
    sb.writeln('⏱️ *Duration*: ${pkg['duration']}');
    sb.writeln('🚘 *Vehicle Type*: ${pkg['vehicle']}');
    sb.writeln('👥 *Passengers*: $_passengersCount Persons');
    sb.writeln('💵 *Package Base Fare*: ₹${(pkg['baseRate'] as double).toStringAsFixed(0)}');
    sb.writeln('\n📍 *Live Pickup Address*: ${pinData['address']}');
    if (pinData['pincode']!.isNotEmpty) {
      sb.writeln('📮 *Pincode*: ${pinData['pincode']}');
    }
    sb.writeln('🗺️ *Live GPS Maps Pin*: https://maps.google.com/?q=${loc.latitude},${loc.longitude}');
    sb.writeln('\n👉 Please confirm pickup date & driver assignment!');

    WhatsAppService.openWhatsApp(phone: '916381029380', message: sb.toString());
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: const Color(0xFF0F172A),
      appBar: AppBar(
        title: const Text('TourO - ஆன்மீக & சுற்றுலா பயணம்', style: TextStyle(fontWeight: FontWeight.bold, fontSize: 16)),
        backgroundColor: const Color(0xFF1E293B),
        foregroundColor: Colors.white,
        actions: [
          IconButton(
            icon: const Icon(Icons.language, color: Colors.cyanAccent),
            tooltip: 'Open Aishlee-Web TourO',
            onPressed: () {
              Navigator.push(
                context,
                MaterialPageRoute(
                  builder: (_) => const WebModuleScreen(title: 'TourO - Aishlee Web Spiritual Tours', modulePath: 'touro'),
                ),
              );
            },
          ),
        ],
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            // Banner
            Container(
              padding: const EdgeInsets.all(16),
              decoration: BoxDecoration(
                gradient: const LinearGradient(
                  colors: [Color(0xFF0369A1), Color(0xFF075985)],
                  begin: Alignment.topLeft,
                  end: Alignment.bottomRight,
                ),
                borderRadius: BorderRadius.circular(20),
                border: Border.all(color: Colors.cyanAccent.withValues(alpha: 0.3)),
              ),
              child: Row(
                children: [
                  const Text('🛕', style: TextStyle(fontSize: 36)),
                  const SizedBox(width: 14),
                  Expanded(
                    child: Column(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: const [
                        Text(
                          'Tamil Nadu Temple & Hill Packages',
                          style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 16),
                        ),
                        Text(
                          'அறுபடைவீடு, இராமேஸ்வரம் & மலைவாசஸ்தல சுற்றுலா',
                          style: TextStyle(color: Colors.cyanAccent, fontSize: 12),
                        ),
                      ],
                    ),
                  ),
                ],
              ),
            ),
            const SizedBox(height: 20),

            const Text('SELECT TOUR PACKAGE (சுற்றுலா பொதியை தேர்வுசெய்க):', style: TextStyle(color: Colors.grey, fontSize: 11, fontWeight: FontWeight.bold)),
            const SizedBox(height: 10),

            ..._tourPackages.keys.map((pkgKey) {
              final item = _tourPackages[pkgKey]!;
              final isSelected = pkgKey == _selectedPackage;
              return GestureDetector(
                onTap: () => setState(() => _selectedPackage = pkgKey),
                child: Container(
                  margin: const EdgeInsets.only(bottom: 12),
                  padding: const EdgeInsets.all(16),
                  decoration: BoxDecoration(
                    color: isSelected ? Colors.cyan.withValues(alpha: 0.15) : const Color(0xFF1E293B),
                    borderRadius: BorderRadius.circular(16),
                    border: Border.all(
                      color: isSelected ? Colors.cyanAccent : Colors.white12,
                      width: isSelected ? 2 : 1,
                    ),
                  ),
                  child: Row(
                    children: [
                      Text(item['icon'] as String, style: const TextStyle(fontSize: 32)),
                      const SizedBox(width: 14),
                      Expanded(
                        child: Column(
                          crossAxisAlignment: CrossAxisAlignment.start,
                          children: [
                            Text(item['title'] as String, style: const TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 13)),
                            const SizedBox(height: 2),
                            Text(item['subtitle'] as String, style: const TextStyle(color: Colors.grey, fontSize: 11)),
                            const SizedBox(height: 4),
                            Text('Duration: ${item['duration']}', style: const TextStyle(color: Colors.cyanAccent, fontSize: 10, fontWeight: FontWeight.bold)),
                          ],
                        ),
                      ),
                      Text('₹${(item['baseRate'] as double).toStringAsFixed(0)}', style: const TextStyle(color: Colors.cyanAccent, fontWeight: FontWeight.bold, fontSize: 14)),
                    ],
                  ),
                ),
              );
            }),

            const SizedBox(height: 16),

            // Passengers Counter
            Container(
              padding: const EdgeInsets.all(16),
              decoration: BoxDecoration(
                color: const Color(0xFF1E293B),
                borderRadius: BorderRadius.circular(16),
                border: Border.all(color: Colors.white12),
              ),
              child: Row(
                mainAxisAlignment: MainAxisAlignment.spaceBetween,
                children: [
                  const Text('Passenger Count (பயணிகள் எண்ணிக்கை):', style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold, fontSize: 13)),
                  Row(
                    children: [
                      IconButton(
                        icon: const Icon(Icons.remove_circle_outline, color: Colors.redAccent),
                        onPressed: () {
                          if (_passengersCount > 1) setState(() => _passengersCount--);
                        },
                      ),
                      Text('$_passengersCount Persons', style: const TextStyle(color: Colors.cyanAccent, fontWeight: FontWeight.bold, fontSize: 14)),
                      IconButton(
                        icon: const Icon(Icons.add_circle_outline, color: Colors.cyanAccent),
                        onPressed: () => setState(() => _passengersCount++),
                      ),
                    ],
                  ),
                ],
              ),
            ),
            const SizedBox(height: 20),

            // 1-Click Package Booking Button
            SizedBox(
              width: double.infinity,
              height: 52,
              child: ElevatedButton.icon(
                onPressed: _bookTourPackage,
                icon: const Icon(Icons.chat, color: Colors.white),
                label: const Text(
                  'BOOK TOUR PACKAGE VIA WHATSAPP (1-CLICK)',
                  style: TextStyle(fontWeight: FontWeight.bold, fontSize: 12, color: Colors.white),
                ),
                style: ElevatedButton.styleFrom(
                  backgroundColor: const Color(0xFF0284C7), // Sky Blue
                  shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
                ),
              ),
            ),
          ],
        ),
      ),
    );
  }
}

File: apps/fago_native/app/src/main/java/com/fago/fagoapp/ui/screens/modules/TourOScreen.kt
package com.fago.fagoapp.ui.screens.modules

import com.fago.fagoapp.auth.AuthUiState
import android.content.Intent
import android.net.Uri
import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material.icons.filled.AddCircleOutline
import androidx.compose.material.icons.filled.Chat
import androidx.compose.material.icons.filled.RemoveCircleOutline
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalContext
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

data class TourPackageItem(
    val key: String,
    val title: String,
    val subtitle: String,
    val icon: String,
    val baseRate: Double,
    val duration: String,
    val vehicle: String
)

/**
 * TourOScreen — 100% parity with Flutter's touro_screen.dart.
 * Features:
 *   - Tamil Nadu Temple & Hill tour packages
 *   - Passenger count counter (+/- persons)
 *   - Live pickup location & pincode
 *   - 1-Click WhatsApp package booking engine to 916381029380
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TourOScreen(
    authState: AuthUiState? = null,
    onBack: () -> Unit
) {
    val context = LocalContext.current

    var selectedPackageKey by remember { mutableStateOf("arupadaiveedu") }
    var passengerCount by remember { mutableIntStateOf(4) }
    var travellerName by remember { mutableStateOf(authState?.fullName ?: "") }
    var travellerPhone by remember { mutableStateOf(authState?.phone ?: "") }

    LaunchedEffect(authState?.fullName, authState?.phone) {
        if (!authState?.fullName.isNullOrBlank()) travellerName = authState.fullName!!
        if (!authState?.phone.isNullOrBlank()) travellerPhone = authState.phone!!
    }

    val tourPackages = listOf(
        TourPackageItem("arupadaiveedu", "அறுபடைவீடு ஆன்மீக பயணம் (Arupadaiveedu)", "Palani, Tiruchendur, Swamimalai, Thiruthani, Madurai", "🕉️", 12500.0, "3 Days / 2 Nights", "Innova / Ertiga / Tempo Traveller"),
        TourPackageItem("rameswaram",    "இராமேஸ்வரம் & கன்னியாகுமரி (Rameswaram)",       "Ramanathaswamy Temple, Dhanushkodi, Vivekananda Rock", "🌊", 14000.0, "3 Days / 2 Nights", "AC SUV / Force Traveller"),
        TourPackageItem("ooty_kodai",    "ஊட்டி & கொடைக்கானல் (Ooty & Kodaikanal)",     "Botanical Garden, Pykara, Pillar Rocks, Coaker Walk",  "⛰️", 16500.0, "4 Days / 3 Nights", "Ghat Road Expert Driver + Force Traveller"),
        TourPackageItem("tanjore_chola", "தஞ்சை பெரிய கோவில் (Tanjore Big Temple)",      "Brihadeeswarar Temple, Gangaikonda Cholapuram",        "🛕", 8500.0,  "2 Days / 1 Night",  "Sedan / SUV / Tempo")
    )

    val selectedPkg = tourPackages.find { it.key == selectedPackageKey } ?: tourPackages[0]

    Scaffold(
        containerColor = Color(0xFF0F172A),
        topBar = {
            TopAppBar(
                title = { Text("🕉️ TourO — ஆன்மீக & சுற்றுலா பயணம்", color = Color(0xFF00F0FF), fontWeight = FontWeight.Bold, fontSize = 16.sp) },
                colors = TopAppBarDefaults.topAppBarColors(containerColor = Color(0xFF1E293B)),
                navigationIcon = {
                    IconButton(onClick = onBack) {
                        Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Back", tint = Color(0xFF00F0FF))
                    }
                }
            )
        }
    ) { padding ->
        LazyColumn(
            modifier = Modifier.padding(padding).fillMaxSize().padding(16.dp),
            verticalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            // Header Banner
            item {
                Surface(
                    shape = RoundedCornerShape(16.dp),
                    color = Color(0xFF0369A1),
                    modifier = Modifier.fillMaxWidth().border(1.dp, Color(0xFF00F0FF).copy(alpha = 0.3f), RoundedCornerShape(16.dp))
                ) {
                    Row(modifier = Modifier.padding(16.dp), verticalAlignment = Alignment.CenterVertically) {
                        Text("🛕", fontSize = 36.sp)
                        Spacer(Modifier.width(14.dp))
                        Column {
                            Text("Tamil Nadu Temple & Hill Packages", color = Color.White, fontWeight = FontWeight.Bold, fontSize = 16.sp)
                            Text("அறுபடைவீடு, இராமேஸ்வரம் & மலைவாசஸ்தல சுற்றுலா", color = Color(0xFF00F0FF), fontSize = 12.sp)
                        }
                    }
                }
            }

            item {
                Text("SELECT TOUR PACKAGE (சுற்றுலா பொதியை தேர்வுசெய்க):", color = Color.Gray, fontSize = 11.sp, fontWeight = FontWeight.Bold)
            }

            // Tour Package List
            items(tourPackages) { item ->
                val isSel = item.key == selectedPackageKey
                Surface(
                    shape = RoundedCornerShape(14.dp),
                    color = if (isSel) Color(0xFF00F0FF).copy(alpha = 0.15f) else Color(0xFF1E293B),
                    modifier = Modifier
                        .fillMaxWidth()
                        .border(
                            width = if (isSel) 2.dp else 1.dp,
                            color = if (isSel) Color(0xFF00F0FF) else Color(0xFF334155),
                            shape = RoundedCornerShape(14.dp)
                        )
                        .clickable { selectedPackageKey = item.key }
                ) {
                    Row(modifier = Modifier.padding(16.dp), verticalAlignment = Alignment.CenterVertically) {
                        Text(item.icon, fontSize = 30.sp)
                        Spacer(Modifier.width(14.dp))
                        Column(modifier = Modifier.weight(1f)) {
                            Text(item.title, color = Color.White, fontWeight = FontWeight.Bold, fontSize = 13.sp)
                            Spacer(Modifier.height(2.dp))
                            Text(item.subtitle, color = Color.Gray, fontSize = 11.sp)
                            Spacer(Modifier.height(4.dp))
                            Text("Duration: ${item.duration}", color = Color(0xFF00F0FF), fontSize = 10.sp, fontWeight = FontWeight.Bold)
                        }
                        Text("₹${item.baseRate.toInt()}", color = Color(0xFF00F0FF), fontWeight = FontWeight.Bold, fontSize = 14.sp)
                    }
                }
            }

            // Passenger Counter
            item {
                Surface(
                    shape = RoundedCornerShape(14.dp),
                    color = Color(0xFF1E293B),
                    modifier = Modifier.fillMaxWidth().border(1.dp, Color(0xFF334155), RoundedCornerShape(14.dp))
                ) {
                    Row(
                        modifier = Modifier.padding(16.dp),
                        horizontalArrangement = Arrangement.SpaceBetween,
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        Text("Passenger Count (பயணிகள் எண்ணிக்கை):", color = Color.White, fontWeight = FontWeight.Bold, fontSize = 13.sp)
                        Row(verticalAlignment = Alignment.CenterVertically) {
                            IconButton(onClick = { if (passengerCount > 1) passengerCount-- }) {
                                Icon(Icons.Default.RemoveCircleOutline, contentDescription = null, tint = Color(0xFFF43F5E))
                            }
                            Text("$passengerCount Persons", color = Color(0xFF00F0FF), fontWeight = FontWeight.Bold, fontSize = 14.sp)
                            IconButton(onClick = { passengerCount++ }) {
                                Icon(Icons.Default.AddCircleOutline, contentDescription = null, tint = Color(0xFF00F0FF))
                            }
                        }
                    }
                }
            }

            // 1-Click WhatsApp Booking Button
            item {
                Button(
                    onClick = {
                        val pName = travellerName.ifBlank { authState?.fullName?.trim() ?: "Pilgrim / Traveller" }
                        val pPhone = travellerPhone.ifBlank { authState?.phone?.trim() ?: "" }
                        if (pPhone.isBlank()) {
                            android.widget.Toast.makeText(context, "Please enter your mobile phone number in profile settings before booking", android.widget.Toast.LENGTH_LONG).show()
                            return@Button
                        }
                        val msg = Uri.encode(
                            "🕉️ *TOURO TAMIL NADU TEMPLE & HILL TOUR BOOKING* 🕉️\n\n" +
                            "👤 *Pilgrim / Traveller Name*: $pName\n" +
                            "📱 *Cell / WhatsApp*: +91 $pPhone\n" +
                            "🚩 *Selected Package*: ${selectedPkg.title}\n" +
                            "⏱️ *Duration*: ${selectedPkg.duration}\n" +
                            "🚘 *Vehicle Type*: ${selectedPkg.vehicle}\n" +
                            "👥 *Passengers*: $passengerCount Persons\n" +
                            "💵 *Package Base Fare*: ₹${selectedPkg.baseRate.toInt()}\n" +
                            "📍 *Pickup Location*: Gandhipuram, Coimbatore 641012, Tamil Nadu\n\n" +
                            "👉 TourO Manager: Please confirm tour booking & driver assignment!"
                        )
                        val waUri = Uri.parse("https://api.whatsapp.com/send?phone=916381029380&text=$msg")
                        context.startActivity(Intent(Intent.ACTION_VIEW, waUri))
                    },
                    modifier = Modifier.fillMaxWidth().height(52.dp),
                    colors = ButtonDefaults.buttonColors(containerColor = Color(0xFF0284C7)),
                    shape = RoundedCornerShape(14.dp)
                ) {
                    Icon(Icons.Default.Chat, contentDescription = null, tint = Color.White)
                    Spacer(Modifier.width(8.dp))
                    Text("BOOK TOUR PACKAGE VIA WHATSAPP (1-CLICK)", color = Color.White, fontWeight = FontWeight.Bold, fontSize = 12.sp)
                }
            }
        }
    }
}
